src/components/Navbar.js

```diff
@@ -17,6 +17,54 @@ export function Navbar() {
   const titleInputRef = useRef(null);
   const [isLoading, setIsLoading] = useState(false);
 
+  const getAllLabels = () => {
+    const allLabels = new Set();
+    Object.values(columns).forEach(column => {
+      column.tasks?.forEach(task => {
+        task.labels?.forEach(label => {
+          allLabels.add(JSON.stringify(label));
+        });
+      });
+    });
+    return Array.from(allLabels).map(label => JSON.parse(label));
+  };
+
+  const handleFilterChange = (filters) => {
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
const handleFilterChange = async (filters) => {
  try {
    await Promise.all(Object.keys(columns).map(async (columnId) => {
      const column = columns[columnId];
      const filteredTasks = column.tasks?.filter(task => {
        // ... filtering logic
      });
      await updateColumn(columnId, { filteredTasks });
    }));
  } catch (error) {
    console.error('Error applying filters:', error);
    // Add user feedback mechanism here
  }
};
```
The handleFilterChange function modifies column data directly without error handling. Add try-catch block to handle potential errors during filtering and column updates.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;